features/steps/black_box_steps.py

```diff
@@ -0,0 +1,261 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+
+"""Module Docstring
+
+We should do as much as we can to document what this module is doing. So let's
+do that here!
+"""
+
+from __future__ import print_function, unicode_literals
+import os
+import time
+
+from behave import given, when, then
+from lxml import etree
+import metsrw
+
+import environment
+from features.steps import utils
+
+
+@given(
+    u'the transfer "DemoTransfer" is started with the automatedProcessingMCP '
+    'processing configuration.')
+def step_impl(context):
+    """Step 1"""
+    if not utils.browse_default_ts_location(context):
+        raise environment.EnvironmentError("Location cannot be verified")
+    try:
+        context.transfer_uuid = utils.start_transfer(context)
+    except environment.EnvironmentError as err:
+        assert False, "Error starting transfer: {}".format(err)
+
+
+@given(
+    u'an AIP has been created and stored')
+def step_impl(context):
+    context.execute_steps(
+        'Given the transfer "DemoTransfer" is started with the'
+        ' automatedProcessingMCP processing configuration.\n'
+        'When the Transfer is COMPLETE\n'
+        'And the Ingest is COMPLETE\n'
+        )
+
+@when(u'the Transfer is COMPLETE')
+def step_impl(context):
+    """Step 2"""
+    context.sip_uuid = None
+    status = None
+    resp = None
+    try:
+        while status != "COMPLETE" or status == "FAILED":
```

Comment:
```suggestion
        while status != "COMPLETE" and status != "FAILED":
```
This is a mistake in logic on my part. If there is a problem at all and the transfer fails, then it will otherwise loop for infinity. Does this suggested line resolve this for if it fails do you think?

Reply:
Yes. I was able to reproduce the infinite loop by turning off ElasticSearch, making AIP indexing fail and the ingest.